Tests/test_Entrez_online.py

```diff
@@ -41,13 +50,20 @@
 
 
 class EntrezOnlineCase(unittest.TestCase):
+    def test_construct_cgi_einfo(self):
+        """Test constructed url for request to Entrez."""
+        cgi = 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/einfo.fcgi'
+        params = Entrez._construct_params(params=None)
+        options = Entrez._encode_options(ecitmatch=False, params=params)
+        result_url = Entrez._construct_cgi(cgi, post=False, options=options)
+        self.assertTrue(result_url.startswith(URL_HEAD + "einfo.fcgi?"), result_url)
+        self.assertTrue(URL_TOOL in result_url)
+        self.assertTrue(URL_EMAIL in result_url)
 
-    def test_read_from_url(self):
+    @patch("Bio.Entrez._open", return_value=_binary_to_string_handle(open("Entrez/einfo1.xml", "rb")))
+    def test_read_from_url(self, mock_open):
```

Comment:
I think it ``mock_open`` is defined and used at runtime. Actually instead of ``mock_open`` it could be any name. The test will run if the argument is present regardless of name, otherwise it will fail. Mock is weird.